datafusion/core/src/datasource/file_format/arrow.rs

```diff
@@ -193,6 +193,13 @@ impl DisplayAs for ArrowFileSink {
     }
 }
 
+/// Initial writing buffer size. Note this is just a size hint for efficiency. It
+/// will grow beyond the set value if needed.
+const INITIAL_BUFFER_BYTES: usize = 1048576;
```

Comment:
Thanks for documenting that, yesterday was pondering what those values exactly mean


there are more to go

https://github.com/apache/arrow-datafusion/blob/main/datafusion/core/src/datasource/file_format/parquet.rs#L1028

https://github.com/apache/arrow-datafusion/blob/main/datafusion/core/src/datasource/file_format/parquet.rs#L1007


https://github.com/apache/arrow-datafusion/blob/main/datafusion/core/src/datasource/file_format/parquet.rs#L683

Reply:
Filed follow on PR https://github.com/apache/arrow-datafusion/pull/8656